packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -0,0 +1,631 @@
+    final int paintId = buffer.getInt32();
+    final int verticesLength = buffer.getInt32();
+    final Float32List vertices = buffer.getFloat32List(verticesLength);
+    final int indexLength = buffer.getInt32();
+    Uint16List? indices;
+    if (indexLength != 0) {
+      indices = buffer.getUint16List(indexLength);
+    }
+    listener?.onDrawVertices(vertices, indices, paintId);
+  }
+}
+
+/// Implement this listener class to support decoding of vector_graphics binary
+/// assets.
+abstract class VectorGraphicsCodecListener {
+  /// A paint object has been decoded.
+  ///
+  /// If the paint object is for a fill, then [strokeCap], [strokeJoin],
+  /// [strokeMiterLimit], and [strokeWidget] will be `null`.
+  void onPaintObject(
+    int color,
+    int? strokeCap,
+    int? strokeJoin,
+    int blendMode,
+    double? strokeMiterLimit,
+    double? strokeWidth,
+    int paintStyle,
+    int id,
+  );
+
+  /// A path object is being created, with the given [id] and [fillType].
+  ///
+  /// All subsequent path commands will refer to this path, until
+  /// [onPathFinished] is invoked.
+  void onPathStart(int id, int fillType);
+
+  /// A path object should move to (x, y).
+  void onPathMoveTo(double x, double y);
+
+  /// A path object should line to (x, y).
+  void onPathLineTo(double x, double y);
+
+  /// A path object will draw a cubic to (x1, y1), with control point 1 as
+  /// (x2, y2) and control point 2 as (x3, y3).
+  void onPathArcTo(
+      double x1, double y1, double x2, double y2, double x3, double y3);
+
+  /// The current path has been closed.
+  void onPathClose();
+
+  /// The current path is completed.
+  void onPathFinished();
+
+  /// Draw the given [pathId] with the given [paintId].
+  ///
+  /// If the [paintId] is `null`, a default empty paint should be used instead.
+  void onDrawPath(
+    int pathId,
+    int? paintId,
+  );
+
+  /// Draw the vertices with the given [vertices] and optionally index buffer
+  /// [indices].
+  ///
+  /// If the [paintId] is `null`, a default empty paint should be used instead.
+  void onDrawVertices(
+    Float32List vertices,
+    Uint16List? indices,
+    int? paintId,
+  );
+}
+
+enum _DecodePhase {
+  paints,
+  paths,
+  commands,
+}
+
+/// Write-only buffer for incrementally building a [ByteData] instance.
+///
+/// A [VectorGraphicsBuffer] instance can be used only once. Attempts to reuse will result
+/// in [StateError]s being thrown.
+///
+/// The byte order used is [Endian.host] throughout.
+class VectorGraphicsBuffer {
+  /// Creates an interface for incrementally building a [ByteData] instance.
+  VectorGraphicsBuffer()
+      : _buffer = Uint8Buffer(),
+        _isDone = false,
+        _eightBytes = ByteData(8) {
+    _eightBytesAsList = _eightBytes.buffer.asUint8List();
+    // Begin message with the magic number and current version.
+    _putUint32(VectorGraphicsCodec._magicNumber);
+    _putUint8(VectorGraphicsCodec._version);
+  }
+
+  Uint8Buffer _buffer;
```

Comment:
I'm not a huge fan of the implementation of this. I'd rather avoid depending on package:typed_data if we can.

Reply:
fair enough, I'll rip it out and replace it with something simpler

Reply:
I just replaced this with a List<int> 🤷 

Also fixed the zero padding issue!